### ISSUE TITLE
Taildropping photos that have the same filename produces a corrupted file

### ISSUE THREAD

**Issue by vincentlu** [tailscale/tailscale#2114]
Photos in the iOS, iPadOS, or macOS Photos app library can have the same filename when exported out of the library. When they are sent via Taildrop, there is only one file created in the receiving device, and the content looks garbled.

Steps to reproduce:
(1) In the Photos app (iOS, iPadOS, or macOS), select multiple photos that are known to have the same filename. I'm only aware of one approach of preparing such test files: I use SwiftScan Pro which (when configured) saves a photo in the photo library in addition to creating a PDF file. All photos added to the library this way have the same filename final.JPG. I'll update this issue if I find another way to prepare test files.
(2) Click the share button, select Tailscale, and then select a device (iPhone or iPad or Mac).
(3) Once the operation is completed, check the receiving device, ~/Downloads on macOS or the Tailscale folder in On My iPhone or On My iPad. In my test, there is only one final.JPG, which looks garbled, as if multiple JPG photos were mixed into one.

One approach to address this may be to rename the files automatically. In my test, the 3 final.JPG photos become final.JPG, final 2.JPG, final 3.JPG when "Saved to Files" using the share menu. 

Environment:
macOS 11.4 with Tailscale 1.8.5
iOS/iPadOS 14.6 with Tailscale 1.8.7

**Comment by bradfitz** [comment]
@nickoneill @soniaappasamy @crawshaw ....

Step (1) is probably easy enough to repro with the CLI's `--name` flag:

```
$ tailscale file cp -h
USAGE
  file cp <files...> <target>:

FLAGS
  -name ...       alternate filename to use, especially useful when <file> is "-" (stdin)
  -targets false  list possible file cp targets
  -verbose false  verbose output
```

(I haven't tried)

Note that the Windows file receiving code already does this automatic renaming on collisions.

That the files are "garbled" implies that it's overwriting existing files, without even O_TRUNC.

**Comment by vincentlu** [comment]
I tried sending the same three files from Linux to macOS and iOS:

Linux => macOS
Only one file shows up in ~/Downloads. It's whichever the last file in the file cp command. It's not garbled. In fact, the MD5 sum matches. While the files are being copied, in Finder they are overwritten one after another.

Linux => iOS
Three notifications show up. Clicking each one receives a different file. None were garbled.

$ uname -a
Linux algo 5.4.0-74-generic #83-Ubuntu SMP Sat May 8 02:35:39 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

$ ./tailscale version
date.20210603
  go version: go1.16.5

**Comment by DentonGentry** [comment]
Fixed in 1.18 by https://github.com/tailscale/corp/pull/2854 (which you won't be able to access)
